theano/gpuarray/basic_ops.py

```diff
@@ -863,6 +871,7 @@ class GpuReshape(HideC, tensor.Reshape):
     def make_node(self, x, shp):
         ctx_name = infer_context_name(x)
         x = as_gpuarray_variable(x, context_name=ctx_name)
+        shp = tensor.as_tensor_variable(shp)
```

Comment:
This is a separate crash fix. I have named the commit likewise so that if it is required as a different PR, we can do it